lib/util/include/hse_util/atomic.h

```diff
@@ -86,7 +86,10 @@
 #define atomic_dec_return(_ptr) \
     (atomic_fetch_sub((_ptr), 1) - 1)
 
-#define atomic_cmpxchg(_ptr, _oldp, _new) \
+#define atomic_add_return(_ptr, _val) \
```

Comment:
Oh and shouldn't these be uppercased? :)

Reply:
It predates me. The orginal HSE atomics we're created to have the same names and semantics as the atomics in the Linux kernel so that any HSE code that used them could run in either the kernel or userland.
We could change over completely to using C11 atomics, but that naming doesn't match what we originally started with, so that would take quite a bit of effort to get right.  When I converted to C11 atomics, I changed the underlying implementation to use C11 atomics, but did not change the HSE naming.  And so here we are :-)

Reply:
At some point, maybe we can just move to full C11 atomics and get rid of the extra APIs then, except for cases where you need to change the atomics mode.

Reply:
Agreed, but I suspect you'd fast grow weary of reading and writing:

atomic_load_explicit(&myvar, memory_order_relaxed);

when your intent is:

atomic_read(&myvar)

And it can get much worse, e.g.:

#define atomic_cmpxchg_acq(_ptr, _oldp, _new) \
    atomic_compare_exchange_weak_explicit((_ptr), (_oldp), (_new),      \
                                          memory_order_acquire, memory_order_relaxed)

Reply:
I saw the names aren't amazing, but 🤷